### PR TITLE
Add owner dashboard

### LINKED ISSUE #107: Add Owner Dashboard
There are a new page for cacheowners (https://www.geocaching.com/play/owner) with new lists:
https://www.geocaching.com/play/owner/published
https://www.geocaching.com/play/owner/unpublished
https://www.geocaching.com/play/owner/archived
Here I would add s2cgeo-buttons

### LINKED ISSUE #108: Script crashed, if a cache have a comment
grafik
If there is a comment on a cache, the script crashed.

### PR DESCRIPTION
**Pull request by capoaira**
fix #108 
fix #107

### CODE REVIEW

File: send2cgeo.user.js [hunk 1]
```diff
@@ -311,6 +313,35 @@ function s2cgGCMain() {
...
+            $("#send2cgeo iframe")
+                .hide()
+                .off('load')
+                .attr('src', 'https://send2.cgeo.org/add.html?cache=' + GCCode)
+                .on('load',
+                    function() {
+                        // hide "please wait text" and show iframe
+                        $("#send2cgeo div").hide();
+                        // hide box after 3 seconds
+                        $(this).css('display', 'block').parent().delay(3000).fadeOut();
```
Comment (kumy) [inline send2cgeo.user.js:339]: Codacy found an issue: [Strings must use doublequote. (quotes)](https://app.codacy.com/manual/cgeo/send2cgeo/pullRequest?prid=5837471)

**Comment by kumy** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Clones added
============
- send2cgeo.user.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cgeo/send2cgeo/pullRequest?prid=5837471&bid=19113817)

File: send2cgeo.user.js [hunk 2]
```diff
@@ -311,6 +313,35 @@ function s2cgGCMain() {
...
+
+        $('#s2cg-' + GCCode).on('click', function() {
+            // show the box and the "please wait" text
+            $("#send2cgeo, #send2cgeo div").fadeIn();
+            // hide iframe for now and wait for page to be loaded
+            $("#send2cgeo iframe")
+                .hide()
+                .off('load')
+                .attr('src', 'https://send2.cgeo.org/add.html?cache=' + GCCode)
+                .on('load',
```
Comment (kumy) [inline send2cgeo.user.js:334]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/manual/cgeo/send2cgeo/pullRequest?prid=5837471)

File: send2cgeo.user.js [hunk 3]
```diff
@@ -311,6 +313,35 @@ function s2cgGCMain() {
...
+
+        $(anchorElement).append(html);
+
+        $('#s2cg-' + GCCode).on('click', function() {
+            // show the box and the "please wait" text
+            $("#send2cgeo, #send2cgeo div").fadeIn();
+            // hide iframe for now and wait for page to be loaded
+            $("#send2cgeo iframe")
+                .hide()
+                .off('load')
```
Comment (kumy) [inline send2cgeo.user.js:332]: Codacy found an issue: [Strings must use doublequote. (quotes)](https://app.codacy.com/manual/cgeo/send2cgeo/pullRequest?prid=5837471)

**Comment by kumy** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- send2cgeo.user.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cgeo/send2cgeo/pullRequest?prid=5837471&bid=19113817)

**Comment by capoaira** [comment]
Can someone have a look at this PR?

File: send2cgeo.user.js [hunk 4]
```diff
@@ -327,6 +358,8 @@ function s2cgGCMain() {
 
     // Send to c:geo on seachmap (new map)
     if (document.location.href.match(/\.com\/play\/map/)) {
+        // Remove the pedding for the ul
```
Comment (bekuno) [inline send2cgeo.user.js:361]: Do you mean padding?

**Comment by bekuno** [comment]
The code for the owner part looks fine, but I can not test it.

@Lineflyer Can you test the function on the owner sites?

File: send2cgeo.user.js (hunk 4 above)
Comment (capoaira) [reply send2cgeo.user.js]: Oops

**Comment by kumy** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- send2cgeo.user.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cgeo/send2cgeo/pullRequest?prid=5837471&bid=19113817)

**Comment by Lineflyer** [comment]
@kumy Can you try if you can end the codacy analysis for this repo with your account.

I recently enabled it for cgeo-bot account but seems it also continues in the name of your account. See this PR comments above.

File: send2cgeo.user.js [hunk 5]
```diff
@@ -310,7 +312,36 @@ function s2cgGCMain() {
...
         + '</div>');
 
-// This function add the send2cgeo buttons on geocaching.com
+    // This function add the send2cgeo buttons on geocaching.com
+    // Because jQuery is not supported by some pages, the window.s2geo() function does not work.
+    // The following function is a workaround to solve this problem.
+    function buildButton(GCCode, anchorElement, height, imgClass='') {
+        // Add s2cg button.
+        var html = '<a id="s2cg-' + GCCode + '" href="javascript:void(0);" title="Send to c:geo">'
+            + '<img class="' + imgClass + '" src="https://send2.cgeo.org/send2cgeo.png" height="' + height + '"/>'
```
Comment (bekuno) [inline send2cgeo.user.js:321]: Please move the function upwards direct after the last function.

**Comment by kumy** [comment]
> Can you try if you can end the codacy analysis for this repo with your account.

@Lineflyer  done :)

**Comment by bekuno** [comment]
@Lineflyer Can you test the function on the owner sites?

**Comment by Lineflyer** [comment]
I will merge it to master, test as time permits, report back and merge to ˋreleaseˋ if ok.

**Pull request merged**